steward/src/cellars/cellar_v2.rs

```diff
@@ -185,503 +159,47 @@ fn get_encoded_adaptor_call(data: Vec<AdaptorCall>) -> Result<Vec<AbiAdaptorCall
-                                spender: sp_call_parse_address(p.spender)?,
-                            };
-                            calls.push(UniswapV3AdaptorCalls::RevokeApproval(call).encode().into())
-                        }
-                        uniswap_v3_adaptor::Function::CollectFees(p) => {
-                            let call = steward_abi::uniswap_v3_adaptor::CollectFeesCall {
-                                token_id: string_to_u256(p.token_id)?,
-                                amount_0: string_to_u128(p.amount_0)?.as_u128(),
-                                amount_1: string_to_u128(p.amount_1)?.as_u128(),
-                            };
-                            calls.push(UniswapV3AdaptorCalls::CollectFees(call).encode().into())
-                        }
-                        uniswap_v3_adaptor::Function::PurgeAllZeroLiquidityPositions(p) => {
-                            let call = steward_abi::uniswap_v3_adaptor::PurgeAllZeroLiquidityPositionsCall {
-                                token_0: sp_call_parse_address(p.token_0)?,
-                                token_1: sp_call_parse_address(p.token_1)?,
-                            };
-                            calls.push(
-                                UniswapV3AdaptorCalls::PurgeAllZeroLiquidityPositions(call)
-                                    .encode()
-                                    .into(),
-                            )
-                        }
-                        uniswap_v3_adaptor::Function::PurgeSinglePosition(p) => {
-                            let call = steward_abi::uniswap_v3_adaptor::PurgeSinglePositionCall {
-                                token_id: string_to_u256(p.token_id)?,
-                            };
-                            calls.push(
-                                UniswapV3AdaptorCalls::PurgeSinglePosition(call)
-                                    .encode()
-                                    .into(),
-                            )
-                        }
-                        uniswap_v3_adaptor::Function::RemoveUnownedPositionFromTracker(p) => {
-                            let call = steward_abi::uniswap_v3_adaptor::RemoveUnOwnedPositionFromTrackerCall {
-                                token_id: string_to_u256(p.token_id)?,
-                                token_0: sp_call_parse_address(p.token_0)?,
-                                token_1: sp_call_parse_address(p.token_1)?,
-                            };
-                            calls.push(
-                                UniswapV3AdaptorCalls::RemoveUnOwnedPositionFromTracker(call)
-                                    .encode()
-                                    .into(),
-                            )
-                        }
-                    }
-                }
+            UniswapV3Calls(params) => {
+                calls.extend(adaptors::uniswap_v3::uniswap_v3_adaptor_v1_call(params)?)
```

Comment:
Extend is then pushing all of the bytes in the result into the `calls` vector of bytes? Was `push(...into())` doing this implicitly before? Just realizing how weird it is considering `calls` should only be one call ever.

Reply:
Before we were looping inside of each enum variant. This does bring to my attention the fact that I named these methods as "call" instead of "calls". They return a vec so I'll rename them. That's also why `extend` is used.